refactor: Simplify `handleNetworkSwitch` utility

Gudahtt opened the pull request:
**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/coding_guidelines/CODING_GUIDELINES.md)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add the appropiate QA label when dev review is completed
    - `needs-qa`: PR requires manual QA.
    - `No QA/E2E only`: PR does not require any manual QA effort. Prior to merging, ensure that you have successful end-to-end test runs in Bitrise. 
    - `Spot check on release build`: PR does not require feature QA but needs non-automated verification. In the description section, provide test scenarios. Add screenshots, and or recordings of what was tested.
5. Add `QA Passed` label when QA has signed off (Only required if the PR was labeled with `needs-qa`)
6. Add your team's label, i.e. label starting with `team-` (or `external-contributor` label if your not a MetaMask employee)

**Description**

The `handleNetworkSwitch` no longer requires the network controller and currency rate controller instances to be passed in. This makes it simpler to use, and reduces the usage of global controller references (which in turn helps reduce the effort of managing breaking changes to those controllers).

**Issue**

This relates to https://github.com/MetaMask/mobile-planning/issues/1015

**Checklist**

* [x] There is a related GitHub issue
* [x] Tests are included if applicable
* [x] Any added code is fully documented

Gudahtt committed:
refactor: Simplify `handleNetworkSwitch` utility

The `handleNetworkSwitch` no longer requires the network controller and
currency rate controller instances to be passed in. This makes it
simpler to use, and reduces the usage of global controller references
(which in turn helps reduce the effort of managing breaking changes to
those controllers).

This relates to https://github.com/MetaMask/mobile-planning/issues/1015

Gudahtt commented on app/util/networks/index.test.ts:
```diff
@@ -182,49 +180,6 @@ describe('NetworkUtils::compareRpcUrls', () => {
   });
 });
 
-describe('NetworkUtils::handleNetworkSwitch', () => {
```
These tests were replaced with a dedicated test module with complete coverage

Gudahtt commented:
Passing e2e test run: https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/5986e5da-d930-4fd7-8073-751e99ccd169

Cal-L commented on app/util/networks/handleNetworkSwitch.ts:
```diff
@@ -1,33 +1,35 @@
-    networkController?.state?.providerConfig?.chainId ===
-    String(switchToChainId)
-  ) {
+  if (networkController.state.providerConfig.chainId === switchToChainId) {
```
Should we keep the `String` conversion in here in case `switchToChainId` is still a number for some reason?

Gudahtt replied:
Great catch, I had meant to review this to see if this was safe. I don't think it is, I found a couple of weird cases at least. I'll restore this.

```suggestion
  if (networkController.state.providerConfig.chainId === String(switchToChainId)) {
```

Gudahtt committed:
Restore String conversion

The parameter should already be a string, but this would prevent an exception in the event the wrong type was passed in.

Gudahtt committed:
Fix lint error

Cal-L reviewed (approved):
LGTM!
